examples/mm/stable_diffusion/train_sd_draftp.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.multiprocessing as mp
+from megatron.core import parallel_state
+from megatron.core.utils import divide
+from omegaconf.omegaconf import OmegaConf, open_dict
+
+from nemo.collections.multimodal.models.text_to_image.stable_diffusion.ldm.ddpm import MegatronLatentDiffusion
+from nemo.collections.nlp.parts.megatron_trainer_builder import MegatronStableDiffusionTrainerBuilder
+from nemo.collections.nlp.parts.peft_config import PEFT_CONFIG_MAP
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager
+from nemo_aligner.algorithms.supervised import SupervisedTrainer
+from nemo_aligner.data.mm import text_webdataset
+from nemo_aligner.data.nlp.builders import build_dataloader
+from nemo_aligner.models.mm.stable_diffusion.image_text_rms import get_reward_model
+from nemo_aligner.models.mm.stable_diffusion.megatron_sd_draftp_model import MegatronSDDRaFTPModel
+from nemo_aligner.utils.distributed import Timer
+from nemo_aligner.utils.train_script_utils import (
+    CustomLoggerWrapper,
+    add_custom_checkpoint_callback,
+    extract_optimizer_scheduler_from_ptl_model,
+    init_distributed,
+    init_using_ptl,
+    retrieve_custom_trainer_state_dict,
+)
+
+mp.set_start_method("spawn", force=True)
+
+
+@hydra_runner(config_path="conf", config_name="draftp_sd")
+def main(cfg) -> None:
+
+    logging.info("\n\n************** Experiment configuration ***********")
+    logging.info(f"\n{OmegaConf.to_yaml(cfg)}")
+
+    # TODO: has to be set true for PyTorch 1.12 and later.
+    torch.backends.cuda.matmul.allow_tf32 = True
+    cfg.model.data.train.dataset_path = [cfg.model.data.webdataset.local_root_path for _ in range(cfg.trainer.devices)]
+    trainer = MegatronStableDiffusionTrainerBuilder(cfg).create_trainer()
+    exp_manager(trainer, cfg.exp_manager)
+    logger = CustomLoggerWrapper(trainer.loggers)
+    # TODO: @geshen: For the Stable Diffusion, I am currently getting the vae and unet separately
+    # TODO: @ataghibakhsh: Redo with aligner style and find out the reason
+    ptl_model = MegatronLatentDiffusion(cfg.model, trainer).to(torch.cuda.current_device())
+    # TODO: @geshen: Check why we have PEFT init here
+    if cfg.model.get("peft", None):
```

Comment:
since this MR we've abstracted this part away https://github.com/NVIDIA/NeMo-Aligner/commit/1e4f4cf20cbca194be60a670ebf7ab3854db2fc4

Reply:
can you see if you can use that PEFT logic?

Reply:
done